cli/tree/src/main/scala/edu/rpi/tw/twxplore/cli/tree/command/EtlCommand.scala

```diff
@@ -15,8 +23,230 @@ object EtlCommand extends Command {
+    val result: String = str.substring(0, startIndex) + str.substring(startIndex+1, endIndex).replace(",", "+") + str.substring(endIndex+1)
+    result
+  }
+
   def apply(): Unit = {
-//    val pipelineName = args.pipelineName.toLowerCase()
+    val pipelineName = args.pipelineName.toLowerCase()
+    val dataDirectoryPath = Paths.get(args.dataDirectoryPath)
+    val source = scala.io.Source.fromFile(args.dataDirectoryPath)
+    var treeList: ListBuffer[Tree] = new ListBuffer[Tree]()
+    var treeSpeciesList: mutable.HashMap[String, TreeSpecies] = new mutable.HashMap()
+    var boroughList: mutable.HashMap[Int, Borough] = new mutable.HashMap()
+    var ntaList: mutable.HashMap[String, NTA] = new mutable.HashMap()
+    var blockList: mutable.HashMap[Int, Block] = new mutable.HashMap()
+    var postalCode: mutable.HashMap[Int, Postcode] = new mutable.HashMap()
+    val state: State = State("New York", new ListBuffer[City])
+    val city: City = City("New York City", new ListBuffer[Borough], new ListBuffer[Postcode], state.state)
+    state.cities += city
+    var line_no = 0
+
+    for(line <- source.getLines) {
+      line_no += 1
+      breakable {
+        if(line_no == 1)
+          break
+        val problemStart: Int = line.indexOf("\"")
+        var cols: Array[String] = new Array[String](3)
+        if (problemStart != -1) {
+          val problemEnd: Int = line.indexOf("\"", problemStart + 1)
+          val new_line = replaceComma(line, problemStart, problemEnd)
+          cols = new_line.split(",", -1).map(_.trim)
+        } else {
+          cols = line.split(",", -1).map(_.trim)
+        }
+
+        val treeId: Int = cols(0).toInt
+        val dateFormatter = new java.text.SimpleDateFormat("MM/d/yyyy")
+        val createdAt: Date = dateFormatter.parse(cols(2))
+        val treeDbh: Int = cols(3).toInt
+        val stumpDia: Int = cols(4).toInt
+
+        val curbLoc: CurbLoc = cols(5) match {
+          case "OnCurb" => OnCurb
+          case "OffsetFromCurb" => OffsetFromCurb
+        }
+
+        val status: Status = cols(6) match {
+          case "Alive" => Alive
+          case "Dead" => Dead
+          case "Stump" => Stump
+        }
+
+        val health: Option[Health] = cols(7) match {
+          case "Fair" => Some(Fair)
+          case "Good" => Some(Good)
+          case "Poor" => Some(Poor)
+          case _ => None
+        }
+
+        val treeSpecies: Option[TreeSpecies] = treeSpeciesList.get(cols(8)) match {
+          case Some(t) => Some(t)
+          case _ => {
+            val newTreeSpecies: Option[TreeSpecies] = cols(8) match {
+              case "" => None
+              case _ => {
+                val tempSpecies = Some(TreeSpecies(cols(8), cols(9)))
+                treeSpeciesList += (cols(8) -> tempSpecies.get)
+                tempSpecies
+              }
+            }
+            newTreeSpecies
+          }
+        }
+
+
+
+        val steward: Option[Steward] = cols(10) match {
```

Comment:
All of these should be separate functions in the LineProcessor.
It can then do process(Line): Tree

Then the result is just

Tree(
...
   sidewalk=processSidewalk(col.get(whatever)) 
...
)